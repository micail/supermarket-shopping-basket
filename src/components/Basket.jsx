import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {Button} from 'react-bootstrap';

import { addItem } from '../redux/actions/receiptActions';


const Basket = () => {
  const dispatch = useDispatch();
  const priceList = useSelector((state) => state.priceList);
  const buyItem = (position) => {
    dispatch(addItem(position));
  };

  return (
    <div className="Basket col-md-6 d-flex flex-column">
      {
        priceList.map((item, index) => {
          const itemName = (item[0]).toUpperCase();
          const itemPrice = item[1];
          const position = index;
          return (
            <Button className="m-1" variant="success" size="md" key={itemName} type="button" onClick={() => buyItem(position)}>
              {itemName}
              {' '}
              {itemPrice}
            </Button>
          );
        })
      }
    </div>
  );
};

export default Basket;
