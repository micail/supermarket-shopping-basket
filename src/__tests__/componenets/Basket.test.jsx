import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import Basket from '../../components/Basket';

const mockStore = configureMockStore();

describe('<Basket />', () => {
  const store = mockStore({
    priceList: [['Beans', 0.50]],
    receipt: [],
    totals: {},
  });
  const wrapper = mount(<Provider store={store}><Basket /></Provider>);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render buttons based on list price items', () => {
    expect(wrapper.find('button').text()).toEqual('BEANS 0.5');
  });
});
