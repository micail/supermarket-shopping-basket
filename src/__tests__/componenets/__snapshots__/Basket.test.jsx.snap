// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Basket /> should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Basket>
    <div
      className="Basket"
    >
      <button
        key="BEANS"
        onClick={[Function]}
        type="button"
      >
        BEANS
         
        Â£0.5
      </button>
    </div>
  </Basket>
</Provider>
`;
